content/en/docs/setup/upgrade/index.md

```diff
@@ -1,10 +1,52 @@
 ---
-title: Upgrade Istio using istioctl
-description: Upgrade or downgrade Istio using the istioctl upgrade command.
+title: Upgrade Istio
+description: Upgrade or downgrade Istio.
 weight: 25
 keywords: [kubernetes,upgrading]
 ---
 
+## Canary upgrades
+
+The recommended upgrade method is to do a canary deployment of the control plane,
+and slowly migrate traffic to the new version. This allows making safe changes
+and monitor the impact on the mesh before moving all traffic over.
+
+To support canary deployments, Istio has an installation setting named `revision`.
+Using this option, you can deploy independent control planes.
+Each revision will have its own Deployment, Service, etc.
+
+### Control plane
+
+To install a new revision, named `canary` for example, set the `revision` field:
+`istioctl install --set revision=canary`.
+
+After this runs, see there are two deployments running side-by-side:
+
+{{< text bash >}}
+$ kubectl get pods -n istio-system
+NAME                                    READY   STATUS    RESTARTS   AGE
+istiod-786779888b-p9s5n                 1/1     Running   0          114m
+istiod-canary-6956db645c-vwhsk          1/1     Running   0          1m
+{{< /text >}}
+
+### Data plane
+
+Simply installing the new revision has no impact on the existing proxies. To upgrade these,
+configure them to point to the new control plane. This is controlled during sidecar injection
+based on the namespace label `istio.io/rev`. To upgrade the namespace `test-ns`, relabel it
+to point to the canary revision: `kubectl label namespace default istio-injection- istio.io/rev=canary`. Note the removal
```

Comment:
```suggestion

{{< text bash >}}
$ kubectl label namespace default istio-injection- istio.io/rev=canary
{{< /text >}}

Note that the above command is also removing the `istio-injection` label because otherwise
it would take precedence over the `istio.io/rev` label (for backward compatibility).
```